File: 백준/Silver/20300. 서강근육맨/서강근육맨.js
const path = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(path).toString().trim().split("\n");

let [n, ...loss] = input;
n = +n;
loss = loss[0].split(' ').map(i => BigInt(i));

function solution(n, loss) {
  let answer = 0;
  loss.sort((a, b) => a < b ? -1 : 1);

  if (n % 2 === 1) {
    answer = loss.pop();
    n -= 1;
  }

  for (let i = 0; i < n / 2; i += 1) {
    const sum = loss[i] + loss[loss.length -1 - i];
    answer = sum > answer ? sum : answer;
  }

  return answer;
}

const answer = solution(n, loss);
console.log(String(answer));

File: 백준/Silver/20300. 서강근육맨/서강근육맨.py
# 20300 서강근육맨

n = int(input())
m = 0
muscle_loss = list(map(int, input().split()))

muscle_loss.sort(reverse=True)

if len(muscle_loss) % 2 != 0:
    last = muscle_loss[0]
    muscle_loss = muscle_loss[1:]
    m = last

while len(muscle_loss) > 0:
    maximum = muscle_loss[0]
    minimum = muscle_loss[-1]

    if maximum + minimum > m:
        m = maximum + minimum

    muscle_loss = muscle_loss[1:]
    muscle_loss = muscle_loss[0:-1]

print(m)
